File: python/bob/bob.py
"""Bob - a lackadaisical teenager answering bot.

Answers:
 - 'Sure.' if questioned, such as "How are you?".
 - 'Whoa, chill out!' if YELLED AT (in all capitals).
 - 'Calm down, I know what I'm doing!' if yelled a question.
 - 'Fine. Be that way!' if not saying anything.
 - 'Whatever.' otherwise.
"""


def response(hey_bob: str) -> str:
    """Answer according to given prompt.

    :param hey_bob: the prompt bob should answer
    :return: Bob's answer to the prompt
    """
    prompt = hey_bob.strip()
    if not prompt:
        return "Fine. Be that way!"

    question = True if prompt.endswith("?") else False
    yell = True if prompt.isupper() else False
    if question and yell:
        return "Calm down, I know what I'm doing!"

    if question:
        return "Sure."

    if yell:
        return "Whoa, chill out!"

    return "Whatever."

File: javascript/bob/bob.js
/*
* Answer as Bob, according to given prompt.
*
* @param {string} message prompt to answer
* @returns {boolean} Bob's answer
*/
export const hey = (message) => {
  const msg = message.trim();
  if(msg === '') return "Fine. Be that way!";

  const question = msg[msg.length -1] === "?"
  const yell = msg.toUpperCase() === msg && msg !== msg.toLowerCase()
  if (question && yell) {
    return "Calm down, I know what I'm doing!";
  } else if (question) {
    return 'Sure.';
  } else if (yell) {
    return 'Whoa, chill out!';
  }
  return 'Whatever.';
};
